msteams-platform/messaging-extensions/api-based-secret-service-auth.md

```diff
@@ -0,0 +1,137 @@
+
+1. Under **API key name**, add a meaningful name for the API Key. For example, API key for Contoso message extension.
+
+1. Under **Base URL**, specify a path that initiates all the API endpoints. The path must start with https, include a fully qualified domain name, and optionally, a path. Teams only transmits the secrets to endpoints where the URL begins with this value. For example, `https://api.yelp.com`. *[Mandatory]*
+
+   Base URL ensures that the key remains secure and isn't leaked to random endpoints, even if another app illicitly acquires the API secret registration ID and incorporates it into their own app. We enforce this URL constraint on API keys. If the path registered here doesn't prefix the target endpoint defined in the app manifest, the call gets dropped.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-register-key-domain.png" alt-text="Screenshot shows the Description and Add domain options in the API key registration page in Developer Portal for Teams.":::
+
+1. Under **Target tenant**, select any of the following:
+
+   * **Home tenant**: The API key is only functional within the tenant where it's registered.
+   * **Any tenant**: The API key t is usable in any tenant.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-api-key-tenant.png" alt-text="Screenshot shows the Home tenant and Any tenant options under set a target tenant heading in Developer Portal for Teams.":::
+
+1. Under **Target Teams app**, select any of the following:
+
+   * **Existing Teams app**: The **Existing Teams app** option binds the API secret registration to your specific Teams app.
+   * **Any Teams app**: The API key can be used with any Teams app.
+
+   Adding a domain ensures that the key isn't exposed to endpoints. However, the API secret registration ID is publicly accessible and can be added to random apps, potentially sending unwanted data to a developer's endpoint. To prevent this, you can bind the registration to a specific app and Teams rejects requests for any app other than the one specified in the secret registration.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-api-key-teams-app.png" alt-text="Screenshot shows the Any Teams app and Existing Teams app options under Set a Teams app heading in Developer Portal for Teams.":::
+
+   An **API key registration ID** is generated.
+
+   :::image type="content" source="../assets/images/Copilot/api-based-me-api-key-reg-id.png" alt-text="Screenshot shows the API key registration ID generated in Developer Portal for Teams.":::
+
+1. In Developer portal for Teams, select **Apps** and select an app where you want to add the API key.
+
+1. Go to **App features** > **Message extension**.
+
+1. Under **Authentication**, select **API key** and add the API key registration ID.
```

Comment:
It's weird that I don't see Authentication section in my TDP. I'm following up with Mao to confirm if this is fully rolled out.

Reply:
@leonyang12 , please let us know once you confirm with Mao.

Reply:
This is rolled out:
![image](https://github.com/user-attachments/assets/68bf48d3-a0fe-41b8-84ec-0470cc40180d)